plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsBlobContainer.java

```diff
@@ -63,6 +63,11 @@
     // TODO: See if we can get precise result reporting.
     private static final DeleteResult DELETE_RESULT = new DeleteResult(1L, 0L);
 
+    @Override
+    public boolean blobExists(String blobName) throws IOException {
+        return store.execute(fileContext -> fileContext.util().exists(new Path(path, blobName)));
```

Comment:
This one we used to do differently in https://github.com/original-brownbear/elasticsearch/commit/4b8fd4e76f1e344d8994486f28c96d950303cf1a#diff-a6d76133025d0cd3d4c12918d42be05bL66 ... but just catching any io exception and returning `false` seemed broken to me so I didn't bring that back. Especially since this may now trigger marking a repository as corrupted which we don't want to happen on e.g. a network issue with HDFS.